src/dmd/parse.d

```diff
@@ -1515,6 +1515,36 @@ final class Parser(AST) : Lexer
         }
     }
 
+    private bool isTypeCtors()
```

Comment:
I think this should be `isTypeCtor` without the trailing 's'.

Reply:
Is there some way you can consolidate this function and https://github.com/dlang/dmd/blob/55ca5920ca0f7da1f4096eafb95feaffa6813a79/src/dmd/parse.d#L1518-L1552 to remove the reduandancy an duplication?

Reply:
I put a s because it's for a pack of TypeCtor. Actually if you have just one an any bad token following the func returns false. It's really to detect the pack that can be found before a BasicType.

Reply:
I'm aware that it can look redundant but the problem is that parseTypeCtor works on the current token while isTypeCtor is a lookup function.

Reply:
What about renaming `isTypeCtor()` something like `advanceToTypeCtor` and moving the `case` statement to a ` bool isTypeCtor(TOK value)`?  Then `advanceToTypeCtor` and `parseTypeCtor` can both call `isTypeCtor`.  Or something like that.

Reply:
Better yet, we should be able to pull out the whole `while` loop into it's own function.  I can't work out the details right now, but D's such a powerful language.  There's got to be a way.

Reply:
I can simplify the lookup. Just look at next token. If it's a type ctor that's not followed by a left paren then call parseTypeCtor. This will work.

Reply:
No that didn't work